atomic_reactor/plugins/pre_resolve_composes.py

```diff
@@ -105,13 +105,8 @@ def run(self):
             self.wait_for_composes()
         except WaitComposeToFinishTimeout as e:
             self.log.info(str(e))
-            preserve_composes = set(self.compose_ids).union(self.parent_compose_ids)
-            cancel_composes = set(self.all_compose_ids) - preserve_composes
-            if cancel_composes:
-                self.log.info('Canceling unfinished composes which were created by the build: %s',
-                              cancel_composes)
 
-            for compose_id in cancel_composes:
```

Comment:
I feel like this may have been intentional. Not 100% sure though, the plugin updates `all_compose_ids` so many times that I've lost track of what it's supposed to represent at this point in the code

Reply:
I *think* that:
- compose_ids is the list of IDs that the user specified when starting the build
- parent_compose_ids is the list of compose IDs from the parent build
- all_compose_ids is compose_ids + parent_compose_ids + IDs for composes that OSBS requested for this build

So the cancel_composes list probably *was* what we wanted - the list of IDs for composes that were requested in this build

But I would appreciate if someone could double-check my theory, I'm still confused by this plugin

Reply:
yeah your logic is correct that is exactly how it works

so by removing that part of code, you are also checking status of composes finished already in the past, so I'd just leave that code here and check only really those newly requested ones

Reply:
1 note/bug is that in case that compose_ids were provided AND they required renewal, we aren't canceling those

but that is because in 'wait_for_composes' it is looping through them and requesting renewal if required, and once loop finishes it updates all_compose_ids

so maybe it would be best just create something like self.new_compose_ids
and add there ids in 'request_compose_if_needed' and 'wait_for_composes' (if self._needs_renewal)

that way we will collect all newly requested ones and we can use just that

Reply:
got it. Thanks for the suggestion @rcerven . I have followed your suggestion and updated the logic accordingly.